contracts/src/deployers/reth/RETHHyperdriveDeployerCoordinator.sol

```diff
@@ -0,0 +1,126 @@
+    IRocketStorage public immutable rocketStorage;
+
+    /// @dev The Rocket Token RETH contract.
+    IRocketTokenRETH internal immutable rocketTokenReth;
+
+    /// @notice Instantiates the deployer coordinator.
+    /// @param _coreDeployer The core deployer.
+    /// @param _target0Deployer The target0 deployer.
+    /// @param _target1Deployer The target1 deployer.
+    /// @param _target2Deployer The target2 deployer.
+    /// @param _target3Deployer The target3 deployer.
+    /// @param _target4Deployer The target4 deployer.
+    /// @param _rocketStorage The Rocket Storage contract.
+    constructor(
+        address _coreDeployer,
+        address _target0Deployer,
+        address _target1Deployer,
+        address _target2Deployer,
+        address _target3Deployer,
+        address _target4Deployer,
+        IRocketStorage _rocketStorage
+    )
+        HyperdriveDeployerCoordinator(
+            _coreDeployer,
+            _target0Deployer,
+            _target1Deployer,
+            _target2Deployer,
+            _target3Deployer,
+            _target4Deployer
+        )
+    {
+        rocketStorage = _rocketStorage;
+
+        // Fetching the RETH token address from the storage contract.
+        address rocketTokenRethAddress = _rocketStorage.getAddress(
+            keccak256(abi.encodePacked("contract.address", "rocketTokenRETH"))
+        );
+        rocketTokenReth = IRocketTokenRETH(rocketTokenRethAddress);
+    }
+
+    /// @dev Prepares the coordinator for initialization by drawing funds from
+    ///      the LP, if necessary.
+    /// @param _hyperdrive The Hyperdrive instance that is being initialized.
+    /// @param _lp The LP that is initializing the pool.
+    /// @param _contribution The amount of capital to supply. The units of this
+    ///        quantity are either base or vault shares, depending on the value
+    ///        of `_options.asBase`.
+    /// @param _options The options that configure how the initialization is
+    ///        settled.
+    /// @return value The value that should be sent in the initialize
+    ///         transaction.
+    function _prepareInitialize(
+        IHyperdrive _hyperdrive,
+        address _lp,
+        uint256 _contribution,
+        IHyperdrive.Options memory _options
+    ) internal override returns (uint256 value) {
+        // If base is the deposit asset, revert because depositing as base
+        // is not supported for the rETH integration.
+        if (_options.asBase) {
+            revert IHyperdrive.UnsupportedToken();
+        }
+        // Otherwise, transfer vault shares from the LP and approve the
+        // Hyperdrive pool.
+        rocketTokenReth.transferFrom(_lp, address(this), _contribution);
+        rocketTokenReth.approve(address(_hyperdrive), _contribution);
+
+        return value;
+    }
+
+    /// @dev Allows the contract to receive ether.
+    function _checkMessageValue() internal view override {}
```

Comment:
We should make this prevent payments since we don't have ETH as a deposit asset.